server/opts.go

```diff
@@ -83,8 +89,36 @@ func WithHTTPMux(mux *chi.Mux) Option {
 	}
 }
 
+// WithHTTPRouterMux func sets HTTPMux Router
 func WithHTTPRouterMux(mux transport.Router) Option {
 	return func(o *serverOpts) {
 		o.HTTPMux = mux
 	}
 }
+
+// WithHTTPServer sets HTTP Server to use insted of the default
+func WithHTTPServer(srv *http.Server) Option {
+	if srv == nil {
+		panic("sent Server pointer is nil")
+	}
+	return func(o *serverOpts) {
+		o.HTTPServer = srv
+	}
+}
+
+// WithGracefull applies Gracefull shutdown func to server
+func WithGracefull(fn func(sc chan os.Signal) func() error) Option {
+	if fn == nil {
+		panic("sent Gracefull func is nil")
+	}
+	return func(o *serverOpts) {
+		o.GracefullFunc = fn
+	}
+}
+
+// WithHost sets default server host
+func WithHost(host string) Option {
```

Comment:
It's not immediately clear what `WithHost` option is for. Maybe `ListenHost` or smth would look better?

Reply:
With Host opt func sets host to listen to with port. It's as described in comment.
It is not always good to listen to on 0.0.0.0. What to do if you want to listen on 127.0.0.1 or other host?

Reply:
WithHost is chosen to be complient with other code. All opt func are started from `With`

Reply:
I've meant that `WithListenHost` would look better than `WithHost`, since the latter can mean more than one thing.